factory-package-news/factory-package-news.py

```diff
@@ -27,6 +29,17 @@ def utf8str(content):
     return str(content, 'utf-8') if isinstance(content, bytes) else content
 
 
+# Older versions wrote uncompressed files, support both for now.
+def openZstdOrPlain(path: str):
+    f = open(path, 'rb')
+    magic = f.read(4)
```

Comment:
If it is impossible that a valid file is < 4 byte then this is ok. Otherwise I'd check the file size first.

Reply:
At least here it is